xmodule/services.py

```diff
@@ -308,3 +312,81 @@ def _handle_deprecated_progress_event(self, block, event):
         # in order to avoid duplicate work and possibly conflicting semantics.
         if not getattr(block, 'has_custom_completion', False):
             self.completion_service.submit_completion(block.scope_ids.usage_id, 1.0)
+
+
+class ShowAnswerService(Service):
+    """
+    An XBlock Service that allows XModules to define if the answers are visible.
+    """
+    @classmethod
+    def answer_available(
+            cls, show_answer='', show_correctness='', past_due=False, attempts=0,
+            is_attempted=False, is_correct=False, required_attempts=0,
+            max_attempts=0, used_all_attempts=False, closed=False,
```

Comment:
`is_attempted`, `used_all_attempts`, and `closed` have relatively simple logic, so we could add them as helper methods to this service instead of explicitly passing them from an XBlock.